src/vm/comthreadpool.cpp

```diff
@@ -266,15 +267,15 @@ FCIMPL0(FC_BOOL_RET, ThreadPoolNative::NotifyRequestComplete)
     {
         HELPER_METHOD_FRAME_BEGIN_RET_0();
 
+        if (needReset)
```

Comment:
It gave better throughput, I'm not 100% convinced by it as was moving things around that were showing up as hot - so is an experimental change rather than one with a well thought out reason.

If I was to guess, its because `needReset` is set before `shouldAdjustWorkers` so its result is available earlier in the pipeline? 

Could probably get the same effect by swapping `bool needReset = ` and `bool shouldAdjustWorkers = ` except `shouldAdjustWorkers` is memory fenced.